src/components/Facility/LocationManagement.tsx

```diff
@@ -268,14 +283,21 @@ const Location = ({
         >
           {description || "-"}
         </p>
-        <p className="mt-3 text-sm font-semibold text-secondary-700">
+        <span className="mt-3 text-sm font-semibold text-secondary-700">
           Middleware Address:
-        </p>
+        </span>
+        {!middleware_address && facilityMiddleware && (
+          <span className="ml-2 mt-2 text-xs h-fit rounded-full border-2 border-primary-500 bg-primary-100 px-3 py-[3px]">
```

Comment:
Make use of the badge component (src/components/ui) here instead of custom logic.

As for the design feedback, I'd wait for design team to respond.